File: src/programmers/internal/json_schema_title.ts
import { CommentFactory } from "../../factories/CommentFactory";

import { IJsDocTagInfo } from "../../schemas/metadata/IJsDocTagInfo";

export const json_schema_title = (schema: {
  description?: string | null | undefined;
  jsDocTags?: IJsDocTagInfo[] | undefined;
}): string | undefined => {
  const info: IJsDocTagInfo | undefined = schema.jsDocTags?.find(
    (tag) => tag.name === "title",
  );
  return !!info?.text?.length ? CommentFactory.merge(info.text) : undefined;
};

File: packages/typia/native/core/programmers/iterate/json_schema_title.go
package iterate

import (
  "strings"

  nativemetadata "github.com/samchon/typia/packages/typia/native/core/schemas/metadata"
)

func json_schema_title(schema struct {
  description *string
  jsDocTags   []nativemetadata.IJsDocTagInfo
}) *string {
  for _, tag := range schema.jsDocTags {
    if tag.Name != "title" || len(tag.Text) == 0 {
      continue
    }
    parts := make([]string, 0, len(tag.Text))
    for _, part := range tag.Text {
      parts = append(parts, strings.ReplaceAll(part.Text, "\r\n", "\n"))
    }
    output := strings.Join(parts, "")
    return &output
  }
  return nil
}
